libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -343,6 +361,14 @@ void ContractCompiler::appendFunctionSelector(ContractDefinition const& _contrac
 	}
 
 	FunctionDefinition const* fallback = _contract.fallbackFunction();
+
+	solAssert(!_contract.isLibrary() || !fallback, "Libraries can't have fallback functions");
+
+	bool noPayableFunctions = !hasPayableFunctions(fallback, interfaceFunctions);
```

Comment:
`hasPayableFunctions(_contract)` might be easier to read and also harder to get wrong if we make any changes in the future.